### PR TITLE
[AIP-217] Enforce the unreachable field.

### PR DESCRIPTION
**Pull request by lukesneeringer**

### CODE REVIEW

File: rules/aip0217/synonyms_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+		problems  testutils.Problems
+	}{
+		{"unreachable", testutils.Problems{}},
+		{"unreachable_locations", testutils.Problems{{Suggestion: "unreachable"}}},
+	} {
+		t.Run(strcase.UpperCamelCase(test.FieldName), func(t *testing.T) {
+			f := testutils.ParseProto3Tmpl(t, `
+				message ListBooksResponse {
+					repeated Book books = 1;
+					string next_page_token = 2;
```
Comment (mingzhi) [inline rules/aip0217/synonyms_test.go]: Should we test the `OnlyIf` to gain 100% coverage?

File: docs/rules/0217/unreachable-field-type.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+## Examples
+
+**Incorrect** code for this rule:
+
+```proto
+// Incorrect.
+message ListBooksResponse {
+  repeated Book books = 1;
+  string next_page_token = 2;
+  string unreachable_locations = 3;  // Should be repeated.
```
Comment (mingzhi) [inline docs/rules/0217/unreachable-field-type.md]: Should it be "unreachable" field name?

File: rules/aip0217/unreachable_field_type.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/locations"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var unreachableFieldType = &lint.FieldRule{
+	Name: lint.NewRuleName(217, "unreachable-field-type"),
+	OnlyIf: func(f *desc.FieldDescriptor) bool {
+		return f.GetName() == "unreachable"
```
Comment (mingzhi) [inline rules/aip0217/unreachable_field_type.go]: Do we need to ensure that the message contains"next_page_token"?

File: rules/aip0217/synonyms.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/locations"
+	"github.com/jhump/protoreflect/desc"
+)
+
+var synSet = stringset.New(
+	"unreachable_locations",
+)
+
+var synonyms = &lint.MessageRule{
```
Comment (mingzhi) [inline rules/aip0217/synonyms.go]: Can you use a `FieldRule`? We add a `OnlyIf` function like
```
func(f *desc.FieldDescriptor) bool {
  return f.GetOwner().FindFieldByName("next_page_token") != nil
}
```

File: rules/aip0217/unreachable_field_type.go (hunk 3 above)
Comment (lukesneeringer) [reply rules/aip0217/unreachable_field_type.go]: Nah.

File: docs/rules/0217/unreachable-field-type.md (hunk 2 above)
Comment (lukesneeringer) [reply docs/rules/0217/unreachable-field-type.md]: Good catch, thanks.

Comment (lukesneeringer) [reply docs/rules/0217/unreachable-field-type.md]: Done.

File: rules/aip0217/synonyms_test.go (hunk 1 above)
Comment (lukesneeringer) [reply rules/aip0217/synonyms_test.go]: Done.

File: rules/aip0217/synonyms.go (hunk 4 above)
Comment (lukesneeringer) [reply rules/aip0217/synonyms.go]: Done.

**Pull request merged**